File: 4247-MinimumCapacityBox/4247-MinimumCapacityBox.cpp
// Last updated: 3/30/2026, 2:10:28 PM
class Solution {
public:
    int minimumIndex(vector<int>& capacity, int itemSize) {
        map<int,int> hashmap;
        for(int index = 0 ; index < capacity.size() ; index++){
            if(hashmap.find(capacity[index]) != hashmap.end()) continue; 
            hashmap[capacity[index]] = index;
        }
        for(auto val : hashmap){
            if(val.first >= itemSize) return val.second;
        }
        return -1;
    }
};

File: 4247-MinimumCapacityBox/4247-MinimumCapacityBox.java
// Last updated: 3/30/2026, 2:10:23 PM
class Solution {
    public int minimumIndex(int[] capacity, int itemSize) {
        int n = capacity.length;
        int mincapacity = 101,minindex = -1;
        for(int index = 0 ; index < n ; index++){
            if(capacity[index] >= itemSize && capacity[index] < mincapacity){
                minindex = index;
                mincapacity = capacity[index];
            }
        }
        return minindex;
    }
}